snap_plugin/v1/metric.py

```diff
@@ -100,15 +100,8 @@ def __init__(self, namespace=[], version=None, tags={}, config={},
             raise TypeError("The 'tags' kwarg requires a dict of strings. "
                             "(given: `{}`)".format(type(tags)))
         # configs
-        if isinstance(config, (list, tuple)):
-            self._config = ConfigMap(pb=self._pb.Config, *config)
-        elif isinstance(config, dict):
-            self._config = ConfigMap(pb=self._pb.Config, **config)
-        elif isinstance(config, ConfigMap):
-            self._config = ConfigMap(config.items(), pb=self._pb.Config)
-        else:
-            raise TypeError("The 'config' kwarg requires a list, tuple or"
-                            " dict.  (given: `{}`)".format(type(config)))
+        self._set_config(config)
```

Comment:
Error should be unhandled, as it should be passed to user's scope to let him know he called constructor with wrong args. Added docstring info about errors raised in 2a3ced9 for better readability.